GameServer/gameutils/PlayerTradeWindow.cs

```diff
@@ -616,13 +616,24 @@ public bool AcceptTrade()
 
 						bool tradeSuccess = false;
 
+                        InventoryItem itemtoadd = item;
+
+                        // If PLayer is not Infiltrator (9), Shadowblade (23), Nightshade (49), remove Envenom bonus before add the item in the inventory
+                        if (item.PoisonSpellID > 0 && !(partner.CharacterClass.ID == 9 || partner.CharacterClass.ID == 23 || partner.CharacterClass.ID == 49))

```

Comment:
Just drop the envenom from weapon regardless of class for consistency. It would be abused either way.

Reply:
Fixed